cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -716,8 +716,14 @@ func (s *Supervisor) runAgentProcess() {
 			// https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21079
 
 			// Wait 5 seconds before starting again.
-			restartTimer.Stop()
+			if !restartTimer.Stop() {
+				select {
+				case <-restartTimer.C: // Try to drain the channel
+				default:
+				}
+			}
 			restartTimer.Reset(5 * time.Second)
+			s.commander.ResetDone()
```

Comment:
Could you explain why this is necessary? The channel should still be good to use, and is recreated in the commander's `Start` method anyway.

Reply:
> is recreated in the commander's Start method anyway

True, but it's recreated only after five seconds and without reset it keeps getting triggered.

Reply:
How would they be different from one another? It's not clear to me how splitting `doneCh` into two channels helps.

Reply:
If we split `doneCh` into two channels, we can use each channel for communicating with a particular goroutine. There are two goroutines that currently read from `doneCh`: the one that runs `Commander.Stop` and the other one here in `Supervisor.runAgentProcess`. Splitting this into two channels would let us send messages to each goroutine instead of closing the channel as a way of broadcasting to both of them. If we don't have to close the channels and leave them open, the `select` statement here would be waiting for another message instead of automatically selecting a closed channel.

Reply:
I just re-read the commander code and it makes sense. Let me do that.